### DISCUSSION TITLE
API resolved without sending a response for /XXXX/XXXX, this may result in stalled requests.

### DISCUSSION THREAD

**Discussion by anonymouscatcher** [vercel/next.js#36859] [Help]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

```
    Operating System:
      Platform: darwin
    Binaries:
      Node: 16.13.0
      npm: 8.7.0
      Yarn: 1.22.10
      pnpm: N/A
    Relevant packages:
      next: 12.0.8
      react: 17.0.2
      react-dom: 17.0.2
```

### Describe the Bug

I'm trying to setup local proxy while it's working fine for all the requests, it doesn't work for POST request including data. I have disabled `bodyParser` but it didn't help. 

### Expected Behavior

I expect to resolve the api with a response.

### To Reproduce

Create a local api in `/pages/api/test/[...index].js`. and add the following code and try a post request, for instance an auth api.

```js
const proxy = createProxyMiddleware({
      logLevel: 'debug',
      target: 'XXX',
      changeOrigin: true,
      autoRewrite: true,
    });

export default proxy;
```

NOTE: If you run without next (just a js file and run with node) will work fine and will work as expected.

**Comment by balazsorban44** [comment]
You likely forgot to end the response with `res.end()` or similar in `createProxyMiddleware`.

Could you post the full code?

**Reply by anonymouscatcher** [reply]
This is the full code...the request and response will be passed to function,

```js
createProxyMiddleware(req, res, options);
```

**Reply by anonymouscatcher** [reply]
it only happens while posting a data. I'm sure its an issue related to `bodyParser`. I don't know why next team added this to discussions.

**Reply by anonymouscatcher** [reply]
@balazsorban44 Can you look into this discussion? Is bodyParser has changed? because it seems that disabling it, is not working any more 
https://github.com/vercel/next.js/discussions/11036

**Reply by cdeutsch** [reply]
~@alirezavlz any luck resolving this?~

~I'm running into the same issue.~

UPDATE: never mind. Disabling the `bodyParser` fixed the requests timing out.
https://github.com/vercel/next.js/discussions/11036

I still get the same warning message as you though:
> API resolved without sending a response for [URL], this may result in stalled requests.
